di-ipv-evcs-stub/lambdas/src/handlers/evcsHandler.ts

```diff
@@ -0,0 +1,169 @@
+  const userId = event.pathParameters?.userId;
+  if (!userId) {
+    return buildApiResponse({ errorMessage: "Missing userId." }, 400);
+  }
+
+  try {
+    let request;
+    let res: ServiceResponse = {
+      response: Object,
+    };
+    switch (event.httpMethod) {
+      case "POST":
+        try {
+          request = parsePostRequest(event);
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        } catch (error: any) {
+          console.error(error);
+          return buildApiResponse({ errorMessage: error.message }, 400);
+        }
+        res = await processPostUserVCsRequest(
+          decodeURIComponent(userId),
+          request,
+        );
+        break;
+      case "PATCH":
+        try {
+          request = parsePatchRequest(event);
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        } catch (error: any) {
+          console.error(error);
+          return buildApiResponse({ errorMessage: error.message }, 400);
+        }
+        res = await processPatchUserVCsRequest(
+          decodeURIComponent(userId),
+          request,
+        );
+        break;
+      case "GET": {
+        let accessTokenVarified;
+        try {
+          accessTokenVarified = await verifyAccessToken(
+            validateAccessToken(event.headers?.Authorisation),
+          );
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        } catch (error: any) {
+          console.error(error);
+          return buildApiResponse({ errorMessage: error.message }, 400);
+        }
+
+        if (accessTokenVarified)
+          res = await processGetUserVCsRequest(decodeURIComponent(userId));
+        break;
+      }
+      default:
+        return buildApiResponse(
+          { errorMessage: "Not received correct request http method." },
+          400,
+        );
+    }
+
+    return buildApiResponse(res.response, res.statusCode);
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  } catch (error: any) {
+    console.error(error);
+    return buildApiResponse({ errorMessage: error.message }, 500);
+  }
+}
+
+function parsePostRequest(event: APIGatewayProxyEvent): PostRequest {
+  console.info(`---Request parsing----`);
+  if (!event.body) {
+    throw new Error("Missing request body");
+  }
+
+  const postRequest = JSON.parse(event.body);
+  if (
+    !postRequest ||
+    !postRequest.persistVCs ||
+    postRequest.persistVCs.length <= 0 ||
+    !isValidCreateVcState(postRequest.persistVCs)
+  ) {
+    throw new Error("Invalid request");
+  }
+
+  return postRequest;
+}
+
+function parsePatchRequest(event: APIGatewayProxyEvent): PatchRequest {
+  console.info(`---Request parsing----`);
+  if (!event.body) {
+    throw new Error("Missing request body");
+  }
+
+  const patchRequest = JSON.parse(event.body);
+  if (
+    !patchRequest ||
+    !patchRequest.updateVCs ||
+    patchRequest.updateVCs.length <= 0 ||
```

Comment:
Can these be collapsed into one? 

Something like `if (!patchRequest?.updateVCs?.length <= 0){...}` then usually it will return undefined if any of the lookups return null instead of giving an NPE

Reply:
refactored. ----will like  
patchRequest?.updateVCs?.length <= 0